### PR TITLE
rewrite hierarchy's `getAncestors`

### PR DESCRIPTION
**Pull request by BykhovDenis**

### CODE REVIEW

File: packages/core/src/hierarchy.ts [hunk 1]
```diff
@@ -92,4 +91,18 @@ export class Hierarchy {
...
+    let cl: Ref<Class<Obj>> | undefined = _class
+    while (cl !== undefined) {
+      const list = this.ancestors.get(_class)
+      if (list === undefined) {
+        this.ancestors.set(_class, [cl])
+      } else {
+        list.push(cl)
+      }
+      const attrs = this.classes.get(cl)
+      cl = attrs?.extends
```
Comment (haiodo) [inline packages/core/src/hierarchy.ts:105]: What if A extends B
And user add B and when A?

We should probably check all available descendants on moment of adding A?

Comment (BykhovDenis) [reply packages/core/src/hierarchy.ts:105]: If user try add class A extends class B when B is not exist will be throw error `const attrs = this.classes.get(cl)` here, and I think this is normal

Comment (BykhovDenis) [reply packages/core/src/hierarchy.ts:105]: And we can not check descendants because we use ancestors for create descendants

**Review by haiodo — APPROVED** [review]

**Comment by haiodo** [comment]
Do we have any tests for ascendants?

**Review by z2sx — APPROVED** [review]

**Pull request merged**